common/src/test/java/org/broadleafcommerce/common/sitemap/service/SiteMapGeneratorTest.java

```diff
@@ -133,9 +138,26 @@ protected void testGenerator(List<SiteMapGeneratorConfiguration> smgcList, SiteM
     }
 
     protected void compareFiles(File file1, String pathToFile2) throws IOException {
-        String actualOutput = convertFileToString(file1);
-        String expectedOutput = convertFileToString(new File(pathToFile2));
-        Assert.assertTrue(actualOutput.equals(expectedOutput));
+        try {
+            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+            dbf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+            dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+            dbf.setNamespaceAware(true);
+            dbf.setCoalescing(true);
+            dbf.setIgnoringElementContentWhitespace(true);
+            dbf.setIgnoringComments(true);
+            DocumentBuilder db = dbf.newDocumentBuilder();
+
+            String actualOutput = convertFileToString(file1);
+            String expectedOutput = convertFileToString(new File(pathToFile2));
+            Document doc1 = db.parse(new ByteArrayInputStream(actualOutput.getBytes()));
+            Document doc2 = db.parse(new ByteArrayInputStream(expectedOutput.getBytes()));
+            doc1.normalizeDocument();
+            doc2.normalizeDocument();
+            Assert.assertTrue(doc1.isEqualNode(doc2));
+        } catch (Exception e) {
+            throw new IOException();
```

Comment:
What you are doing here will mask what the actual exception is. Instead of catching all Exceptions and converting them to IOException, please just throw the actual exception in the method signature.